services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -344,6 +344,67 @@ async def remove_users_manually_marked_as_guests(
         )
 
 
+async def _remove_single_orphaned_service(
```

Comment:
why so much modifications in the garbage collector? I'm not sure I understand why?

Reply:
There are a few changes, let me go through them:

- if a user looses connectivity and the GC kicks in, it will now:
   - remove all services in parallel (before it was sequential), important for users with bigger studies
   - make sure to not if one of the services had an error, replaced `asyncio.gather` with `logged_gather`
- errors are handled when requesting services from `director-v2`, in case of service degradation a warning message will be shown